mcp/common/src/main/java/org/springframework/ai/mcp/SyncMcpToolCallback.java

```diff
@@ -92,7 +93,8 @@ public ToolDefinition getToolDefinition() {
 		return DefaultToolDefinition.builder()
 			.name(this.prefixedToolName)
 			.description(this.tool.description())
-			.inputSchema(ModelOptionsUtils.toJsonString(this.tool.inputSchema()))
+			.inputSchema(
+					JsonSchemaUtils.ensureValidInputSchema(ModelOptionsUtils.toJsonString(this.tool.inputSchema())))
 			.build();
```

Comment:
The logic for constructing ToolDefinition in both the synchronous and asynchronous sections is almost identical. It is recommended to extract a common helper method to facilitate future maintenance.